bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ScanAdapter.java

```diff
@@ -137,56 +138,97 @@ public void throwIfUnsupportedScan(Scan scan) {
 
   /** {@inheritDoc} */
   @Override
-  public ReadRowsRequest.Builder adapt(Scan scan, ReadHooks readHooks) {
+  public void adapt(Scan scan, ReadHooks readHooks, Query query) {
     throwIfUnsupportedScan(scan);
 
-    ReadRowsRequest.Builder requestBuilder = ReadRowsRequest.newBuilder()
-        .setRows(toRowSet(scan))
-        .setFilter(buildFilter(scan, readHooks).toProto());
+    toByteStringRange(scan, query);
```

Comment:
Added `toByteStringRange()` after following Igor's instructions provided in the mail